codegens/csharp-restsharp/lib/restsharp.js

```diff
@@ -124,9 +125,14 @@ module.exports = {
 
             //  snippet to create request in csharp-restsharp
             snippet = '';
+        self.getOptions().forEach((option) => {
+            if (_.isUndefined(options[option.id])) {
```

Comment:
now the option[option.id] can be defined but be of a wrong type. I know we did not check this before but it could be a problem in future. Initially we might have assumed that we can never get wrong type, since it is tightly coupled with the UI. But now since we are trying to develop this independently, we could face this issue.
for example: if followRedirect could be a string, in that case this could be a problem. 
We can discuss more on this.

Reply:
@umeshp7 The only thing we should check for is that our code doesn't throw exceptions for wrong option types.
Approving PR for now